core/src/main/java/io/camunda/connector/impl/secret/SecretHandler.java

```
@@ -52,8 +52,17 @@ public class SecretHandler implements SecretElementHandler, SecretContainerHandl
 
   protected final SecretProvider secretProvider;
 
+  protected Function<String, String> secretReplacer;
+
   public SecretHandler(final SecretProvider secretProvider) {
     this.secretProvider = secretProvider;
+    secretReplacer =
+        name ->
+            Optional.ofNullable(secretProvider.getSecret(name))
+                .orElseThrow(
+                    () ->
+                        new IllegalArgumentException(
```

One last nit, could you throw a custom exception here? This way we could catch it and know we dont expose any information and map it properly. Something like like `SecretException` or `SecretMappingException`?